File: backend-beeyond/src/main/java/at/htl/beeyond/dto/CustomApplicationDto.java
package at.htl.beeyond.dto;

import at.htl.beeyond.entity.ApplicationStatus;
import at.htl.beeyond.entity.CustomApplication;
import at.htl.beeyond.entity.User;
import at.htl.beeyond.validation.Checks;
import javax.validation.GroupSequence;
import javax.validation.constraints.NotBlank;

@GroupSequence({CustomApplicationDto.class, Checks.TemplateContent.class})
public class CustomApplicationDto extends ApplicationDto {

    @NotBlank
    private String content;

    public CustomApplicationDto(Long id, String note, ApplicationStatus status, UserDto owner, String content) {
        super(id, note, status, owner);
        this.content = content;
    }

    public CustomApplicationDto() {
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content.trim();
    }

    public CustomApplication map(User owner) {
        return new CustomApplication(note, owner, content);
    }

    public static CustomApplicationDto map(CustomApplication customApplication) {
        return new CustomApplicationDto(
                customApplication.getId(),
                customApplication.getNote(),
                customApplication.getStatus(),
                UserDto.map(customApplication.getOwner()),
                customApplication.getContent()
        );
    }
}

File: backend-beeyond/src/main/java/at/htl/beeyond/dto/CustomApplicationDto.kt
package at.htl.beeyond.dto

import at.htl.beeyond.entity.ApplicationStatus
import at.htl.beeyond.entity.CustomApplication
import at.htl.beeyond.validation.Checks
import at.htl.beeyond.validation.ValidKubernetes
import java.time.LocalDate
import java.time.LocalDateTime
import javax.validation.GroupSequence
import javax.validation.constraints.NotBlank

@GroupSequence(CustomApplicationDto::class, Checks.KubernetesContent::class)
@ValidKubernetes(groups = [Checks.KubernetesContent::class])
class CustomApplicationDto(
    id: Long? = null,
    note: String? = null,
    status: ApplicationStatus? = null,
    owner: UserDto? = null,
    createdAt: LocalDateTime? = null,
    startedAt: LocalDateTime? = null,
    finishedAt: LocalDateTime? = null,
    content: String? = null,
    namespace: String? = null,
    schoolClass: String? = null,
    toDate: LocalDate? = null,
    purpose: String? = null
) : ApplicationDto(
    id,
    note,
    status,
    owner,
    createdAt,
    startedAt,
    finishedAt,
    namespace,
    schoolClass,
    toDate,
    purpose
) {
    constructor(customApplication: CustomApplication) : this(
        customApplication.id,
        customApplication.note,
        customApplication.status,
        UserDto(customApplication.owner),
        customApplication.createdAt,
        customApplication.startedAt,
        customApplication.finishedAt,
        customApplication.content,
        customApplication.namespace.namespace,
        customApplication.schoolClass,
        customApplication.toDate,
        customApplication.purpose
    )

    @field:NotBlank
    override var content: String? = content
        set(value) {
            if (value != null) {
                field = value.trim()
            }
        }
}
